cwm_worker_tests/upload_download_test.py

```diff
@@ -102,21 +136,39 @@ def run(self):
                 print(f'{context} | Iteration # {iteration_no} | Started')
                 objects = client.list_objects(self.bucket, recursive=True)
                 num_downloaded_files = 0
+                downloads = []
                 for obj in objects:
+                    download_start_time = datetime.datetime.now()
                     object_name = obj.object_name
-                    object_filepath = f'/tmp/{object_name}.{self.threadid}' 
-                    client.fget_object(self.bucket, object_name, object_filepath)
-                    num_downloaded_files += 1
-                    os.remove(object_filepath)
+                    object_filepath = f'/tmp/{object_name}.{self.threadid}'
+                    error = None
+                    try:
+                        client.fget_object(self.bucket, object_name, object_filepath)
+                        num_downloaded_files += 1
+                        object_size_bytes = os.path.getsize(object_filepath)
+                        os.remove(object_filepath)
+                    except S3Error as e:
+                        error = str(e)
+                    except Exception as e:
+                        error = str(e)
+                    download_end_time = datetime.datetime.now()
+                    download_elapsed_time = (download_end_time - download_start_time).total_seconds()
+                    downloads.append({
```

Comment:
this will cause huge unlimited memory usage the more download iterations we use

please write to file directly without keeping the data in RAM like in the example I linked to in this comment - https://github.com/CloudWebManage/cwm-worker-cluster/issues/19#issuecomment-957634681